.github/workflows/docs.yml

```diff
@@ -0,0 +1,109 @@
+name: Build Documentation
+on:
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      cache:
+        description: 'Use build cache'
+        required: false
+        default: true
+  schedule:
+    - cron: "0 4 * * *"
+  push:
+    tags:
+      - "*"
+    branches:
+      - main
+
+jobs:
+  doc:
+    name: Build Documentation
+    runs-on: ubuntu-latest
+    env:
+      PYVISTA_OFF_SCREEN: 'True'
+      ALLOW_PLOTTING: true
+      SHELLOPTS: 'errexit:pipefail'
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install OSMesa VTK
+        run: |
+          sudo apt-get install libosmesa6-dev libgl1-mesa-dev
+          pip install https://github.com/pyvista/pyvista-wheels/raw/add_v9_1_0/vtk-osmesa-9.1.0-cp38-cp38-linux_x86_64.whl
+      - name: Install PyVista
+        run: pip install -e .
+      - name: Install Dependencies and PyVista
+        run: |
+          sudo apt-get install python3-tk pandoc
+          pip install -r requirements_docs.txt
+      - name: PyVista Report
+        run: python -c "import pyvista;print(pyvista.Report())"
+      - uses: actions/cache@v2
+        if: ${{ (github.event.inputs.cache && github.event_name == 'workflow_dispatch') || github.event_name != 'workflow_dispatch' }}
```

Comment:
This needs to be looked at/tested. Also, how can we have this be `false` on tags?